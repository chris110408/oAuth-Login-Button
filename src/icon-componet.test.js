
import Enzyme, { shallow, render, mount } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({
    adapter: new Adapter()
});

import IconComponent from './icon-component'

describe('<IconComponent/>', () => {
    const defaultProps = {
        icon:'fa-google',
        customizePrefixCls:'oauth-login'
    }

    describe('exist:', () => {
        let wrapper
        beforeEach(() => {
            wrapper = shallow(<IconComponent {...defaultProps}/>);
        });
        test('renders <MyComponent /> component', () => {

            expect(wrapper).toHaveLength(1);
        });
        test('type is span', () => {
            expect(wrapper.type()).toEqual('span')
        })
        test('type has children', () => {
            expect(wrapper.prop('children')).toMatchSnapshot()
        })
    });

    describe('classname', () => {


        test('should have proper classname with default props', () => {
            const props = Object.assign({},defaultProps)
            const icon = mount(<IconComponent {...props}/>);
            expect(icon.find('i').length).toEqual(1);
            expect(icon.find('i').prop('className')).toEqual("oauth-login-icon fa fa fa-google")
        });
        test('should show loading', () => {
            const props = Object.assign({},defaultProps,{isLoading:true})
            const icon = mount(<IconComponent {...props}/>);
            expect(icon.find('i').length).toEqual(1);
            expect(icon.find('i').prop('className')).toEqual(expect.stringContaining('fa  fa-circle-o-notch fa-spin'))
        });
        test('should show special span', () => {
            const props = Object.assign({},defaultProps,{isLoading:true,spinner:'spinner'})
            const icon = mount(<IconComponent {...props}/>);
            expect(icon.find('i').length).toEqual(1);
            expect(icon.find('i').prop('className')).toEqual(expect.stringContaining('fa-spin'))
            expect(icon.find('i').prop('className')).toEqual(expect.stringContaining(' fa  spinner'))
        });
        test('should not show special span if isLoading is false', () => {
            const props = Object.assign({},defaultProps,{isLoading:false,spinner:'spinner'})
            const icon = mount(<IconComponent {...props}/>);
            expect(icon.find('i').length).toEqual(1);
            expect(icon.find('i').prop('className')).toEqual(expect.not.stringContaining('fa-spin'))
            expect(icon.find('i').prop('className')).toEqual(expect.not.stringContaining(' fa  spinner'))
        });
        test('should show prop size', () => {
            const props = Object.assign({},defaultProps,{size:'sm'})
            const icon = mount(<IconComponent {...props}/>);
            expect(icon.find('i').length).toEqual(1);
            expect(icon.find('i').prop('className')).toEqual(expect.stringContaining('oauth-login-icon-sm'))
        });
        test('should show icon-only', () => {
            const props = Object.assign({},defaultProps,{iconOnly:true})
            const icon = mount(<IconComponent {...props}/>);
            expect(icon.find('i').length).toEqual(1);
            expect(icon.find('i').prop('className')).toEqual(expect.stringContaining('icon-only'))
        });

    });







});
