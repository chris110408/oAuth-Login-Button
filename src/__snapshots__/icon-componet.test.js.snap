// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconComponent/> exist: type has children 1`] = `
Array [
  <link
    href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css"
    rel="stylesheet"
  />,
  <i
    className="oauth-login-icon fa fa fa-google"
  />,
]
`;
